text_formatter.go

```diff
@@ -73,10 +82,12 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
 	isColored := (f.ForceColors || isColorTerminal) && !f.DisableColors
 
+	f.Mutex.Lock()
 	timestampFormat := f.TimestampFormat
```

Comment:
Can you share what code generates that race report, or a reproducer?
On Tue, Jan 19, 2016 at 11:10 sp-marcos-quesada <notifications@github.com>
wrote:

> In text_formatter.go
> <https://github.com/Sirupsen/logrus/pull/305#discussion_r50132327>:
>
> > @@ -73,10 +82,12 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
> >  	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
> >  	isColored := (f.ForceColors || isColorTerminal) && !f.DisableColors
> >
> > +	f.Mutex.Lock()
> >  	timestampFormat := f.TimestampFormat
>
> Passing our test suite using --race it always fails at the same point, we
> need to get ride of all race conditions, so, that's why we decide to fork
> the package:
>
> WARNING: DATA RACE
> Read by goroutine 12:
> github.com/Sirupsen/logrus.(
>
> *TextFormatter).Format() [[[READS .TimestampFormat ]]]
> /home/travis/gopath/src/github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/text_formatter.go:76
> <http://github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/text_formatter.go:76>
> +0x42c github.com/Sirupsen/logrus.( <http://github.com/Sirupsen/logrus.(>*
> Entry).Reader()
> /home/travis/gopath/src/
> github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/entry.go:41
> +0xaa
> /httpport.go:145 +0x600
> Previous write by goroutine 9:
> github.com/Sirupsen/logrus.(
>
> *TextFormatter).Format() [[[ASSIGNS f.TimestampFormat =
> DefaultTimestampFormat]]]
> /home/travis/gopath/src/github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/text_formatter.go:77
> <http://github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/text_formatter.go:77>
> +0x44c github.com/Sirupsen/logrus.( <http://github.com/Sirupsen/logrus.(>*
> Entry).Reader()
> /home/travis/gopath/src/
> github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/entry.go:41
> +0xaa
> ....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sirupsen/logrus/pull/305/files#r50132327>.
>